elasticsearch/src/main/java/com/amazon/opendistroforelasticsearch/sql/elasticsearch/planner/logical/ElasticsearchLogicalPlanOptimizerFactory.java

```diff
@@ -41,6 +43,8 @@ public static LogicalPlanOptimizer create() {
         new MergeAggAndIndexScan(),
```

Comment:
This rule may be effect by limit push down. e.g. agg -> limit -> relation. with current rules,
1. agg -> IndexScan
2. IndexAggScan, which is unexpected.